nodejs/week4/spam-detector/index.js

```diff
@@ -0,0 +1,89 @@
+class Email {
+	constructor(subject, body) {
+		this.subject = subject;
+		this.body = body;
+	}
+}
+
+class SpamDetector {
+	badWordCheck(email) {
+		const badWords = ["viagra", "offer", "free", "business proposal"];
+
+		const emailBody = email.body;
+		const emailBodyIntoArray = emailBody.split(" ");
+
+		const badWordsInEmail = emailBodyIntoArray.filter((item) => {
+			return badWords.includes(item.toLowerCase());
+		});
+
+		if (badWordsInEmail.length !== 0) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	subjectCheck(email) {
+		const emailSubject = email.subject;
+
+		if (emailSubject.toLowerCase() === "hello") {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	upperCaseCheck(email) {
+		const emailBody = email.body;
+		const emailBodyIntoArray = emailBody.split("");
```

Comment:
I think you are splitting every character here

Reply:
sorry about that,it was a typo. I corrected it